File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/sheet/modules/BloodViewModel.java
package amagi82.modularcharactersheetcreator.ui.sheet.modules;

import android.databinding.ObservableField;

import amagi82.modularcharactersheetcreator.models.modules.Blood;
import amagi82.modularcharactersheetcreator.models.modules.Module;
import amagi82.modularcharactersheetcreator.ui._base.BaseModuleViewModel;

public class BloodViewModel extends BaseModuleViewModel {
    public final ObservableField<Blood> blood = new ObservableField<>();

    public BloodViewModel(Module module) {
        super(module);
        update(module);
    }

    public void update(Module module) {
        blood.set(module.blood());
    }
}

File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/sheet/modules/BloodViewModel.kt
package amagi82.modularcharactersheetcreator.ui.sheet.modules

import amagi82.modularcharactersheetcreator.models.modules.Blood
import amagi82.modularcharactersheetcreator.models.modules.Module
import amagi82.modularcharactersheetcreator.ui._base.BaseModuleViewModel
import android.databinding.ObservableField

class BloodViewModel(module: Module) : BaseModuleViewModel(module) {
    val blood = ObservableField<Blood>()

    init {
        update(module)
    }

    override fun update(module: Module) {
        blood.set(module.blood)
    }
}
